### ISSUE TITLE
'JSON value of type NSMutableDictionary cannot be converted to NSString' related to appInfo 

### ISSUE THREAD

**Issue by karlr-stripe** [stripe/stripe-react-native#200]
**Describe the bug**
When trying to use the SDK in an iOS app I ran into this error : 
```
JSON value '{
    type = git;
    url = "git+https://github.com/stripe/stripe-react-native.git";
}' of type NSMutableDictionary cannot be converted to NSString
 
+[RCTConvert NSString:]
    RCTConvert.m:59
$s19stripe_react_native9StripeSdkC10initialise6params8resolver8rejecterySo12NSDictionaryC_yypSgcySSSg_AKs5Error_pSgtctF
```

Tracing through this, what happens is : 
we try to set `appInfo` to [identify the plugin](https://stripe.com/docs/building-plugins#setappinfo) in 
https://github.com/stripe/stripe-react-native/blob/b831a18d9ccc5f2b68db2d7c7d94bcf1b1af0f86/ios/StripeSdk.swift#L52-L55 

that code expects `appInfo["url"]` to be a String but it's actually an object as seen in the error message.

The value for this comes from here : 
https://github.com/stripe/stripe-react-native/blob/b831a18d9ccc5f2b68db2d7c7d94bcf1b1af0f86/src/components/StripeProvider.tsx#L28-L33 
The URL field for the appInfo (` url: pjson.repository,` ) is coming from the package.json file. 

Although in the Github project [it looks like that URL is a string](https://github.com/stripe/stripe-react-native/blob/b831a18d9ccc5f2b68db2d7c7d94bcf1b1af0f86/package.json#L43), what I find is that when I actually install the SDK through npm, the package.json that is served in the dependency package has this object instead.
It's a valid format for that file type : https://docs.npmjs.com/cli/v7/configuring-npm/package-json#repository 

```
~  npm install @stripe/stripe-react-native
....
~  cat  node_modules/@stripe/stripe-react-native/package.json | grep -A 4  repository
  "repository": {
    "type": "git",
    "url": "git+https://github.com/stripe/stripe-react-native.git"
  },
  "scripts": {
```

I imagine that the file gets rebuilt on `npm`s side or something similiar.

We should be more robust in how we parse that file in StripeProvider.tsx  — at the very least we should probably do `repository.url || repository` but we might want to make that code overall more aware of the package.json format than it is at the moment.

**Comment by thorsten-stripe** [comment]
Thanks @karlr-stripe I am able to reproduce this, installing via `npm` completely messes up the `package.json` file 😭 
![image](https://user-images.githubusercontent.com/23213994/117857230-21c52d00-b2bf-11eb-9650-1ffddd7ae56e.png)

For the time being, please install via `yarn` or find the `node_modules/@stripe/stripe-react-native/package.json` file and change the repository param to: `"repository": "https://github.com/stripe/stripe-react-native",`

**Comment by vigzmv** [comment]
Can this fix be released as 0.1.1 please. Thanks
